src/utils/config.ts

```diff
@@ -0,0 +1,59 @@
+import * as fs from 'fs';
+
+class ConfigProperties {
+    DefaultColor = '#0099ff';
+    SubjectChannelGroupIDs: Array<string> = [];
+}
+
+export type ConfigKey = keyof ConfigProperties;
+
+export class Config {
+    private static _instance: Config;
+    public Properties!: ConfigProperties;
+    private configPath = './config.json';
+
+    public static get Instance(): Config {
+        if (!this._instance) {
+            this._instance = new Config();
+        }
+
+        return this._instance;
+    }
+
+    constructor() {
+        this.LoadConfig();
+    }
+
+    public SetProperty(property: ConfigKey, value: any) {
```

Comment:
I feel like this value has type and it should be `string & Array<string>`. You can create out of it type to export (idk what to call it maybe `Properties`? or anything else wil do...  This may lead to error in `src/commands/config.ts` with `newValueArray` and `newValueRaw` this should be easily fixable by casting these variable to type mentioned above.

Reply:
I have created a new `ConfigValue` type for `value` property of this function, unfortunately, assigning values to an object indexed via keys always expects value of a logical AND of it's object types, if we extend config properties beyond `string` and `string[]` to `number` for example, the resulting type is `never` (`string` & `string[]` & `number`), so I had to cast the `value` of this function to `never` in an assignment call.